### PR TITLE
Add new `BALANCE_TRANSFER` kind to Transactions for Internal Transfers (Empty Balance)

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/4489

⚠️ Require https://github.com/opencollective/opencollective-api/pull/6460

image

### CODE REVIEW

File: components/SendMoneyToCollectiveBtn.js [hunk 1]
```diff
@@ -58,6 +58,9 @@ class SendMoneyToCollectiveBtn extends React.Component {
       description,
       paymentMethod: { id: paymentMethods[0].id },
       frequency: 'ONETIME',
+      customData: {
+        isBalanceTransfer: true,
```
Comment (Betree) [inline components/SendMoneyToCollectiveBtn.js:62]: Seems a bit hacky to use `customData`, this is not what this field was intended to. I would rather use a dedicated GraphQL param.

Comment (SudharakaP) [reply components/SendMoneyToCollectiveBtn.js]: @Betree : Yes, makes sense. I've introduced the `isBalanceTransfer` as a dedicated parameter. Let me know if all looks good now. 😃

**Comment by Betree** [comment]
> Related API PR: opencollective/opencollective-api#6460

PR description should say **Require** opencollective/opencollective-api#6460. It's important to surface the fact that API must be deployed first.

**Review by Betree — APPROVED** [review]
Code looks good to me, I haven't tested the feature itself

**Pull request merged**